src/megatron/energon/eyaml.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from typing import Any, BinaryIO, Union
+
+import ryml
+
+
+def load_yaml(stream: Union[BinaryIO, bytes]) -> Any:
+    """Load a YAML file from a stream or bytes object using rapidyaml/ryml.
+    This is much faster than the standard yaml library, but we don't
+    support all YAML features.
+
+    Args:
+        stream: A stream or bytes object containing the YAML data.
+
+    Returns:
+        The parsed YAML data as a native Python object.
+    """
+
+    if isinstance(stream, bytes):
+        bytes_data = stream
+    else:
+        bytes_data = stream.read()
+    tree = ryml.parse_in_arena(bytes_data)
+    native_obj = _ryml_tree_to_native(tree)
+
+    return native_obj
+
+
+def _cast_scalar(tree: ryml.Tree, nid: int) -> object:
+    """Convert YAML scalar to bool | int | float | None | str."""
+    memview = tree.val(nid)
+    s = memview.tobytes().decode()
+
+    if tree.is_val_quoted(nid):
+        return s
+
+    s_lower = s.lower()
+
+    if s_lower in {"null", "~"}:
+        return None
+    if s_lower in {"true", "yes"}:
+        return True
+    if s_lower in {"false", "no"}:
+        return False
+    try:
+        if "." in s or "e" in s_lower:
+            return float(s)
+        return int(s)
```

Comment:
Wouldn't it be more efficient to check `s.is_alnum()` for int instead of a raise + except? (raise + except is in general quite expensive). May speed up loading `split.yaml`, but slightly decrease `.info.yaml`